src/lxc/start.c

```diff
@@ -349,6 +349,11 @@ int lxc_set_state(const char *name, struct lxc_handler *handler,
 	struct state_client *client;
 	struct lxc_msg msg = {.type = lxc_msg_state, .value = state};
 
+	/* This function will try to connect to the legacy lxc-monitord state
+	 * server and only exists for backwards compatibility.
+	 */
+	lxc_monitor_send_state(name, state, handler->lxcpath);
```

Comment:
The state clients are the one that should always be notified first since monitord will die a horrible horrible death soon. So if you could just add:

```
if (lxc_list_empty(&handler->state_clients)) {
		TRACE("no state clients registered");
		process_unlock();
                lxc_monitor_send_state(...);
		return 0;
}
```

that would be +1.